Dockerfile.ubi9

```diff
@@ -1,47 +1,324 @@
-FROM registry.access.redhat.com/ubi9/ubi:latest
+FROM registry.access.redhat.com/ubi9/ubi-minimal AS base-builder
+
+ARG MAX_JOBS
+
+ARG PYTHON_VERSION=3.12
+
+RUN --mount=type=cache,target=/root/.cache/microdnf \
+    rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    microdnf --setopt=cachedir=/root/.cache/microdnf --nodocs install -y \
+        jq \
+        tar \
+        kmod \
+        git \
+        gcc-toolset-13 \
+        cmake \
+        ninja-build \
+        tk-devel \
+        zlib-devel \
+        xsimd-devel \
+        lcms2-devel \
+        gflags-devel \
+        libraqm-devel \
+        libwebp-devel \
+        libjpeg-devel \
+        rapidjson-devel \
+        boost1.78-devel \
+        re2-devel \
+        numactl-devel \
+        libtiff-devel \
+        openjpeg2 \
+        openssl-devel \
+        libimagequant-devel \
+        libxcb-devel \
+        zeromq-devel \
+        pkg-config \
+        utf8proc \
+        java-17-openjdk-devel \
+        python${PYTHON_VERSION}-devel \
+        python${PYTHON_VERSION}-pip \
+    && rpm -ivh https://mirror.stream.centos.org/9-stream/CRB/ppc64le/os/Packages/utf8proc-devel-2.6.1-4.el9.ppc64le.rpm && \
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+ENV VENV=/opt/vllm
+ENV PATH=$VENV/bin:/root/.cargo/bin:$PATH
+RUN python${PYTHON_VERSION} -m venv $VENV
+
+RUN curl -sL https://ftp2.osuosl.org/pub/ppc64el/openblas/latest/Openblas_0.3.29_ppc64le.tar.gz | tar xvf - -C /usr/local
```

Comment:
have args for all the versions for all the repositories like openblas, torch, arrow, audio etc..